### PR TITLE
Add new external data template

### PR DESCRIPTION
**Pull request by CEKrause**
**Note when merging: The `nodata` way of importing the data YAML to the template should be applied to the product template as well.**

**Note: An additional PR will be needed to make this external data template align with product-v2 template and the new product header box from https://github.com/GeoscienceAustralia/dea-knowledge-hub/pull/245**

* [ ] I have spellchecked my content using Microsoft Word, Grammarly, or otherwise.
* [ ] If required, update the [DEA Tech Alerts and Changelog][TechAlertsChangelog] and the [DEA Sandbox Updates banner][SandboxUpdatesBanner].

[TechAlertsChangelog]: https://github.com/GeoscienceAustralia/dea-knowledge-hub/blob/main/docs/tech-alerts-changelog/_tech_alerts_changelog.md
[SandboxUpdatesBanner]: https://bitbucket.org/geoscienceaustralia/datakube-apps/src/64c28bbf3d0e019d8940547a22f78b9bfd58d739/clusters/dea-sandbox/sandbox.yaml

### CODE REVIEW

**Comment by benji-glitsos-ga** [comment]
**Preview:**

* https://pr-238-preview.khpreview.dea.ga.gov.au/data/theme/external/
* https://pr-238-preview.khpreview.dea.ga.gov.au/data/product/ga-ausbathytopo/

**Comment by benji-glitsos-ga** [comment]
We might want to add the External products category to the homepage here: @CEKrause 

ksnip_20240607-163409

**Comment by CEKrause** [comment]
> We might want to add the External products category to the homepage here: @CEKrause
> 
> ksnip_20240607-163409

I might leave that change to you @benji-glitsos-ga . I'm not sure I know how to make that modification... But I do think it's worth adding to the cards.

**Comment by benji-glitsos-ga** [comment]
I'm wondering if someone could help me write some brief intro text that explains what "Band specifications" are. It will replace this line of intro text that is currently not very descriptive. @robbibt 

> These are the specifications of the bands used in this product.

**Comment by robbibt** [comment]
Something like this?

> These are the specifications of the bands included in this product. Bands from this product can be inspected and loaded using `datacube` on the [DEA Sandbox](https://knowledge.dea.ga.gov.au/guides/setup/Sandbox/sandbox/) or [NCI](https://knowledge.dea.ga.gov.au/guides/setup/NCI/basics/), or accessed externally using [DEA's Explorer STAC API](https://knowledge.dea.ga.gov.au/guides/setup/gis/stac/).

File: docs/_templates/external-data-v1.rst [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+
+       {% if has_access_data %}
+       .. list-table::
+          :name: access-table
+
+          {% if valid_external_data %}
+          * - **{{ external_data_label }}**
+            - 
+              * `{{ valid_external_data.custom_label or external_data_default_name }} <{{ valid_external_data.link }}>`_
+            - The external page should contain information about how to access the data.
```
Comment (robbibt) [inline docs/_templates/external-data-v1.rst:307]: Access the the external pages is probably more about users being able to learn more about the data vs. purely data access (given we're providing our own data access options via DEA's systems). Some possible alternative wordings:

```suggestion
            - Access additional information about this dataset from the external provider.
```

```suggestion
            - Learn more about this data from the external provider.
```

Comment (benji-glitsos-ga) [reply docs/_templates/external-data-v1.rst:307]: > Learn how to access the data from the external provider.

Comment (robbibt) [reply docs/_templates/external-data-v1.rst]: The "how to access the data" is the bit we don't want to include - we want users to access data using DEA systems, and the external provider's page won't provide details on how to load the data in a way that's relevant to DEA.

The purpose of these links is to allow users to find out more information about the products from the provider, allowing us to make our listings minimal and not reproduce that material.

Comment (benji-glitsos-ga) [reply docs/_templates/external-data-v1.rst]: Ok very good point -- I've now reworded it to

> Learn more about the data from the external provider.

File: docs/data/external-data/ga-srtm/_data.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+tags:
+  - digital elevation model
+  - srtm
+
+# Access
+
+external_data_page:
+  link: https://pid.geoscience.gov.au/dataset/ga/72759
+  custom_label: "See an example of how to use this data in the DEA Notebook: `Extracting contour lines </notebooks/How_to_guides/Contour_extraction/>`_"
```
Comment (benji-glitsos-ga) [inline docs/data/external-data/ga-srtm/_data.yaml:39]: @robbibt  Ive added a link to the relevant DEA Notebook https://knowledge.dea.ga.gov.au/notebooks/How_to_guides/Contour_extraction/
to this product: https://pr-238-preview.khpreview.dea.ga.gov.au/data/external-data/ga-srtm/?tab=access

Comment (robbibt) [reply docs/data/external-data/ga-srtm/_data.yaml:39]: Awesome idea - I'd forgotten we used it in that notebook! Just checking though - did you mean to update the "External data page" link? I really liked the original tile, and I'm not sure we should overload it with too much extra functionality - I reckon this would be better done simply as a new "Code example" tile (matching how we do it on our main DEA product pages).

Comment (benji-glitsos-ga) [reply docs/data/external-data/ga-srtm/_data.yaml]: Totally agree. Done

File: docs/data/external-data/abares-clum-2020/_specifications.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Bands
+
+bands:
+  footnotes: null
+  bands_table:
+    - name: alum_class
+      aliases: []
+      resolution: 50 metres
+      crs: "EPSG:3577"
+      units: Metres
```
Comment (robbibt) [inline docs/data/external-data/abares-clum-2020/_specifications.yaml:10]: Either this, or leave this one empty

```suggestion
      units: Classification
```

File: starter-kits/external-data/_specifications.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Bands
+
+bands:
+  footnotes: null
+  bands_table:
+  - name: example_band_name
+    aliases: [example_alias_1, example_alias_2]
+    resolution: 30 m
+    crs: Multiple (local UTM)
```
Comment (robbibt) [inline starter-kits/external-data/_specifications.yaml:9]: Do we need units here?

File: docs/data/external-data/ga-srtm/_specifications.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# Bands
+
+bands:
+  footnotes: >
+    :sup:`1` The 'Nodata' value is ``-340282346638528859811704183484516925440``, however ocean pixels in ``dem`` and `dem_s` are also assigned values of ``0``.
```
Comment (robbibt) [inline docs/data/external-data/ga-srtm/_specifications.yaml:5]:
```suggestion
    :sup:`1` The 'Nodata' value is ``-340282346638528859811704183484516925440``, however ocean pixels in ``dem`` and ``dem_s`` are also assigned values of ``0``.
```

**Review by robbibt — APPROVED** [review]
This looks fantastic to me - I've reviewed the current content and it looks great, and the template should be flexible enough to handle all of our external datasets. Very happy to get this published!

**Comment by robbibt** [comment]
Hey @benji-glitsos-ga, re the most recent commit: I really prefer the original setup we had with the "Access" tab unless we have a strong reason to change it - the clean user experience of having the same content available in the same consistent location across all our listings feels more important to me than the small bit of simplicity we gain through removing that tab. The "how to access the data" content in particular feels far less useful when separated from the access options it relates to. 

I'd also like to leave us the option to provide more detailed access into under the "Access" tab table like we do on our main product listings - that won't be possible or as nice in the new design.

File: docs/data/theme/external-data/_data.yaml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 title: External data
-description: External data products are not produced by DEA; they are produced by external providers such as a different program of Geoscience Australia or a different organisation entirely. These external data products may assist with the analysis of DEA data.
+description: External data products are not produced by DEA; they are produced by external providers such as a different program of Geoscience Australia or a different organisation entirely. These external data products may assist with the analysis of DEA data. This data hasn't been altered except to make it compatible with DEA's systems.
```
Comment (robbibt) [inline docs/data/theme/external-data/_data.yaml]: Slight wording update - might still be a bit wordy but I think putting the "purpose" up front helps.

> **These external data products may assist with the analysis of DEA data.** External data products are not produced by DEA; they are produced by external providers such as a different program of Geoscience Australia or a different organisation entirely. This data **has not** been altered except to make it compatible with DEA's systems.

File: docs/_templates/external-data-v1.rst [hunk 7]
```diff
@@ -81,7 +82,7 @@
    .. admonition:: External data
       :class: note external-data
    
-      This data product is developed by an external party, and is not a DEA product.
+      This data product is developed by an external party ({{ Data.external_provider }}), and is not a DEA product. The data hasn't been altered except to make it compatible with DEA's systems.
```
Comment (robbibt) [inline docs/_templates/external-data-v1.rst]: Slight wording update to be less colloquial:

> The data **has not** been altered except to make it compatible with DEA's systems.

File: docs/data/theme/external-data/_data.yaml (hunk 6 above)
Comment (benji-glitsos-ga) [reply docs/data/theme/external-data/_data.yaml]: I actually had the same idea - I wrote this in my latest edit:

> This external data is provided because it may assist with the analysis of DEA data. These data products are produced by external parties, and are not DEA products. DEA provides this data without modifications except where needed to make it compatible with our systems.

File: docs/_templates/external-data-v1.rst (hunk 7 above)
Comment (benji-glitsos-ga) [reply docs/_templates/external-data-v1.rst]: Latest:

> This data product is produced by an external party ({{ Data.external_provider }}), and is not a DEA product. DEA provides this data without modifications except where needed to make it compatible with our systems.

File: docs/data/theme/external-data/_data.yaml (hunk 6 above)
Comment (robbibt) [reply docs/data/theme/external-data/_data.yaml]: Sounds great!

**Comment by benji-glitsos-ga** [comment]
I made a few changes based on Norm's feedback:
ksnip_20240709-125325
ksnip_20240709-125354
